bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/TableItem.java

```diff
@@ -1276,14 +1276,6 @@ private static void handleDPIChange(Widget widget, int newZoom, float scalingFac
 	if (!(widget instanceof TableItem tableItem)) {
 		return;
 	}
-	Image[] images = tableItem.images;
-	if (images != null) {
-		for (Image innerImage : images) {
-			if (innerImage != null) {
-				Image.win32_new(innerImage, newZoom);
-			}
-		}
-	}
```

Comment:
Where will be the inner images be notified about the change when this is removed?

Reply:
Earlier, we were changing the handle of the image on the zoom level change. The method win32_new used to call handleDPIchange which used to set the new zoomed handle in the image. As of now, we have completely changed the way this all happens. Now we dont need to reset the handle of the image on the zoom event, but rather when the image is drawn, the appropriate handle is obtained using the win32_getHandle method where the client passes the current zoom. So, we don't need to do anything on the dpiChanged event but rather it is automatically zoomed in the drawing phase when the widget containing the image is redrawn after the dpi changed event. Since image is a resource, it can only be drawn by widgets or GC. And all the clients which consume the handle of the image now use win32_getHandle method instead to obtain the right handle.